src/NServiceBus.AcceptanceTests/Core/PublishSubscribe/When_disabling_publishing.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace NServiceBus.AcceptanceTests.PublishSubscribe
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using Configuration.AdvancedExtensibility;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_disabling_publishing : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_subscribe_to_and_receive_events()
+        {
+            Requires.MessageDrivenPubSub();
+
+            await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithDisabledPublishing>(e => e.When(
+                    c => c.Subscribe<TestEvent>()))
+                .WithEndpoint<PublishingEndpoint>(e => e.When(c =>
+                {
+                    if (c.ReceivedSubscription)
+                    {
+                        return Task.FromResult(true);
+                    }
+
+                    return Task.FromResult(false);
+                }, s => s.Publish(new TestEvent())))
+                .Done(c => c.ReceivedEvent)
+                .Run();
+        }
+
+        [Test]
+        public void Should_throw_when_publishing()
```

Comment:
Not sure if still needed but in the past we’ve said that only one test per class is allowed to having test endpoints that shares the same input queue?

Reply:
If you're saying it should be split, both test cases are under the same logical context of "When publishing is disabled". While the rule might be still relevant, I wonder if in this case, since it's testing exception throwing when a publish is attempted, it would be ok to have the two tests together.

Reply:
> Not sure if still needed but in the past we’ve said that only one test per class is allowed to having test endpoints that shares the same input queue?

I've seen several tests using multiple tests in the same file in the core acceptance tests (the not shared ones). Not really sure about this but I see no issue having two tests here. keep in mind we don't ship this test to downstreams.